docs/manual/other-topics/typescript.md

```diff
@@ -296,12 +293,228 @@ async function doStuffWithUserModel() {
 }
 ```
 
+### Using with `AttributesOf`
+
+`AttributeOf` is a utility type you can use to automatically extract attributes from your class definition.
+It's an alternative solution designed to reduce the boilerplate of the above solution.
+
+Some caveats of this solution:
+
+- it's not possible to use `AttributesOf` for `CreationAttributes`.
+  This example opted for making properties that are not set until saved always optional instead.
+- `AttributesOf` cannot whether a field is a getter, so they are not excluded by default. Use the second parameter to exclude getters.
+  eg. `AttributesOf<User, 'projects'>` will exclude the field "projects" from the attribute list.
+- `AttributesOf` excludes fields that are inherited from `Model`.
+  If one of your attributes shares the same name as one of the properties of `Model`, change its name.
+  Doing this is likely to cause issues anyway.
+
+Here is the same example as above, but using `AttributeOf`.
+
+<!-- NOTE for maintainers: Keep the following code in sync with `/types/test/typescriptDocs/ModelInitWithAttributesOf.ts` to ensure it typechecks correctly. -->
+
+```typescript
+import {
+  Association, DataTypes, HasManyAddAssociationMixin, HasManyCountAssociationsMixin,
+  HasManyCreateAssociationMixin, HasManyGetAssociationsMixin, HasManyHasAssociationMixin, Model,
+  ModelDefined, Optional, Sequelize, AttributesOf
+} from 'sequelize';
+
+const sequelize = new Sequelize('mysql://root:asd123@localhost:3306/mydb');
+
+// 'projects' is excluded as it's not an attribute, it's an association.
+class User extends Model<AttributesOf<User, 'projects'>> {
+  // id can be undefined during creation when using `autoIncrement`
+  declare id?: number;
```

Comment:
Is this `declare` needed?

Reply:
Currently yes, due to https://sequelize.org/v6/manual/model-basics.html#caveat-with-public-class-fields

Also relevant is this https://github.com/sequelize/meetings/issues/16